File: java/amp-commanding/src/main/java/amp/commanding/ICommandReceiver.java
package amp.commanding;

import amp.messaging.MessageException;
import cmf.bus.IDisposable;

/**
 * Created with IntelliJ IDEA.
 * User: jar349
 * Date: 5/1/13
 */
public interface ICommandReceiver extends IDisposable {

    <TCOMMAND> void onCommandReceived(ICommandHandler<TCOMMAND> handler) throws MessageException, IllegalArgumentException;
}

File: dotnet/source/amp.commanding/ICommandReceiver.cs
using System;
using System.Collections.Generic;

namespace amp.commanding
{
    public interface ICommandReceiver : IDisposable
    {
        void ReceiveCommand(ICommandHandler handler);

        void ReceiveCommand<TCommand>(Action<TCommand, IDictionary<string, string>> handler) 
            where TCommand : class;
    }
}
